File: Challenges/Coding World Cup/2023/Day 11 - ENG vs AFG - Lethal Delivery.java
public class Solution {
    static int maximumSquareDivisor(int n) {
        // Write your code here.
        if (n <= 0) {
            return 0; 
        }

        int maxSquareDivisor = 1;

        for (int i = (int) Math.sqrt(n); i > 0; --i) {
            if (n % (i * i) == 0) {
                maxSquareDivisor = i * i;
                break;
            }
        }

        return (int) Math.sqrt(maxSquareDivisor);
    }
}

File: Challenges/Coding World Cup/2023/Day 11 - ENG vs AFG - Lethal Delivery.cpp
int maximumSquareDivisor(int &n)
{
    // Write your code here
    if (n <= 0)
    {
        return 0; 
    }

    int maxSquareDivisor = 1;

    for (int i = sqrt(n); i > 0; --i)
    {
        if (n % (i * i) == 0)
        {
            maxSquareDivisor = i * i;
            break;
        }
    }

    return sqrt(maxSquareDivisor);
}
